### ISSUE TITLE
Add step to AfterStep

### ISSUE THREAD

**Issue by drekka** [cucumber/cucumber-ruby#450]
We are using frank-cucumber to run tests against an iOS app. As this app is heavily animated we need to pause after each step to ensure the animations are finished. However there are a couple of steps where we do not want to pause. We have tried the AfterStep hook however we have a significant issue with it. 

    AfterStep do |scenario|
       ...
    end

The issue is that it does not appear to be passed the text of the step. This makes it impossible to tell what step was being executed. It seems strange to us that an AfterStep hook would be created without being passed the step that was executed. Could we please have a version like this:

    AfterStep do |scenario, step|
       ...
    end

**Comment by mattwynne** [comment]
Sadly, @chewi's hack will not work on 2.0 as things stand. We will aim to support this properly through the API during the 2.0 series. I'll keep the ticket open.

It would be awesome if someone wanted to start a PR with a failing test for this.

**Comment by Carlbc18** [comment]
I am also interested in this functionality.

**Comment by djwgit** [comment]
like to have this back too,   as in 1.3.x
in 2.x, scenario reference in the hook is a different one...  hope you guys could fix it soon, thanks
and it does make sense to be able to access the steps within the hook...

**Comment by mattwynne** [comment]
If you're interested in this, please start with a failing scenario that describes what you want.

**Comment by jdougit** [comment]
@mattwynne, thanks for your reply...

basically, I run cucumber test on jenkins, by default, the output flushed out to jenkins console after a scenario is finished. but I'd like to have the output send to jenkins console immediately, 
- before a step executed, print out the step to console.
- after that, there are more console output during the test code of the step,  
that makes the console output more readable and easier to located an issue if fails)
to have it this way, I print out steps in the Before and AfterStep hooks...  (might not be a common usage for others...)

after reading the cucuber-ruby code, seems, 
- in 1.3.8   the scenario instance in the hook is the "real" scenario class instance, it has raw_steps attributes. 
- but after 2.x,  the scenario passed into hook is not that one anymore...  instead, it is
Cucumber::RunningTestCase::Scenario, and from this, not sure how to access its steps...

I have this below in my env.rb
================================
# env.rb
# don't buffer the console output. get real-time update on jenkins console page
$stdout.sync = true

# setup and tear-down, run before and after each Scenario
Before do |scenario|
  print("Scenario: #{scenario.title}\n")
  $stop_count = scenario.raw_steps.count
  $step_index = 0
  # print out the first step if exists
  if $step_index < $stop_count
    print("    #{scenario.raw_steps[$step_index].keyword}#{scenario.raw_steps[$step_index].name} \n")
  end
end

# after each step
AfterStep do |scenario|
  # print out next step if exists
  $step_index += 1
  if $step_index < $stop_count
    print("    #{scenario.raw_steps[$step_index].keyword}#{scenario.raw_steps[$step_index].name} \n")
  end
end

**Comment by mattwynne** [comment]
Hi @jdougit

Try calling `#test_steps` on that object you get in the block. That should give you what you need.

**Comment by t-morgan** [comment]
Just want to verify what I am seeing with Cucumber 2.1.0 is the same as what others see...

When I have an env.rb file with:
```
AfterStep do |argument|
  puts argument.class
  argument.each { |element| puts element.class }
end
```
And I run a feature
Then the output is:
```
Array

Cucumber::Core::Test::Result::Passed
1 scenario (1 passed)
1 step (1 passed)

```

**Comment by pirelenito** [comment]
@t-morgan I can verify the same behavior with Cucumber 2.1.0.

**Comment by mattwynne** [comment]
Closed by 8b347ca0f84789ea097fcab730c457f2d81529f0
